components/logins/src/engine.rs

```diff
@@ -90,16 +91,16 @@ impl PasswordEngine {
         root_sync_key: &KeyBundle,
         sync_ping: &mut telemetry::SyncTelemetryPing,
     ) -> Result<()> {
-        let global_state: Cell<Option<String>> = Cell::new(self.db.get_global_state()?);
+        // migrate our V1 state - this needn't live for long.
+        self.db.migrate_global_state()?;
         let result = sync_multiple(
             &[&self.db],
-            &global_state,
+            &Cell::new(None),
```

Comment:
This needs to be fixed - while we no longer persist the global state to storage, we do want it persisted in memory between syncs.